src/main/java/org/elasticsearch/index/query/RangeQueryBuilder.java

```diff
@@ -41,385 +53,215 @@
      */
-    public RangeQueryBuilder lte(float to) {
-        this.to = to;
-        this.includeUpper = true;
+    @Override
+    public RangeQueryBuilder boost(float boost) {
+        this.boost = boost;
         return this;
     }
 
     /**
-     * The to part of the range query. Null indicates unbounded.
+     * Gets the boost factor for the query.
      */
-    public RangeQueryBuilder lte(double to) {
-        this.to = to;
-        this.includeUpper = true;
-        return this;
+    public float boost() {
+        return this.boost;
     }
 
     /**
-     * Should the lower bound be included or not. Defaults to <tt>true</tt>.
-     */
-    public RangeQueryBuilder includeLower(boolean includeLower) {
-        this.includeLower = includeLower;
-        return this;
-    }
-
-    /**
-     * Should the upper bound be included or not. Defaults to <tt>true</tt>.
+     * Sets the query name for the filter that can be used when searching for matched_filters per hit.
      */
-    public RangeQueryBuilder includeUpper(boolean includeUpper) {
-        this.includeUpper = includeUpper;
+    public RangeQueryBuilder queryName(String queryName) {
+        this.queryName = queryName;
         return this;
     }
 
     /**
-     * Sets the boost for this query.  Documents matching this query will (in addition to the normal
-     * weightings) have their score multiplied by the boost provided.
+     * Gets the query name for the query.
      */
-    @Override
-    public RangeQueryBuilder boost(float boost) {
-        this.boost = boost;
-        return this;
+    public String queryName() {
+        return this.queryName;
     }
 
     /**
-     * Sets the query name for the filter that can be used when searching for matched_filters per hit.
+     * In case of date field, we can adjust the from/to fields using a timezone
      */
-    public RangeQueryBuilder queryName(String queryName) {
-        this.queryName = queryName;
+    public RangeQueryBuilder timeZone(String timezone) {
+        this.timeZone = timezone;
         return this;
     }
 
     /**
-     * In case of date field, we can adjust the from/to fields using a timezone
+     * In case of format field, we can parse the from/to fields using this time format
      */
-    public RangeQueryBuilder timeZone(String timezone) {
-        this.timeZone = timezone;
+    public RangeQueryBuilder format(String format) {
+        this.format = format;
         return this;
     }
 
     @Override
     protected void doXContent(XContentBuilder builder, Params params) throws IOException {
         builder.startObject(RangeQueryParser.NAME);
-        builder.startObject(name);
-        builder.field("from", from);
-        builder.field("to", to);
+        builder.startObject(fieldname);
+        if (this.from instanceof BytesRef) {
+            builder.field("from", ((BytesRef) this.from).utf8ToString());
+        } else {
+            builder.field("from", from);
+        }
```

Comment:
we are going to have this type of `if` in quite some places , maybe we could add a utility method to the base class or even add such a method to XContentBuilder ? Thoughts?